runtime/stdlib/range.go

```diff
@@ -0,0 +1,148 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package stdlib
+
+import (
+	"fmt"
+
+	"github.com/onflow/cadence/runtime/errors"
+	"github.com/onflow/cadence/runtime/interpreter"
+	"github.com/onflow/cadence/runtime/sema"
+)
+
+// InclusiveRangeConstructorFunction
+
+const inclusiveRangeConstructorFunctionDocString = `
+ Constructs a Range covering from start to end.
+ 
+ The step argument is optional and determines the step size. 
+ If not provided, the value of +1 or -1 is used based on the values of start and end. 
+ `
+
+var inclusiveRangeConstructorFunctionType = func() *sema.FunctionType {
+	typeParameter := &sema.TypeParameter{
+		Name:      "T",
+		TypeBound: sema.IntegerType,
+	}
+
+	typeAnnotation := sema.NewTypeAnnotation(
+		&sema.GenericType{
+			TypeParameter: typeParameter,
+		},
+	)
+
+	return &sema.FunctionType{
+		TypeParameters: []*sema.TypeParameter{
+			typeParameter,
+		},
+		Parameters: []sema.Parameter{
+			{
+				Label:          sema.ArgumentLabelNotRequired,
+				Identifier:     "start",
+				TypeAnnotation: typeAnnotation,
+			},
+			{
+				Label:          sema.ArgumentLabelNotRequired,
+				Identifier:     "end",
+				TypeAnnotation: typeAnnotation,
+			},
+			{
+				Identifier:     "step",
+				TypeAnnotation: typeAnnotation,
+			},
+		},
+		ReturnTypeAnnotation: sema.NewTypeAnnotation(
+			&sema.InclusiveRangeType{
+				MemberType: typeAnnotation.Type,
+			},
+		),
+		RequiredArgumentCount: sema.RequiredArgumentCount(2),
```

Comment:
On `master` this has changed:
```suggestion
		// `step` parameter is optional
		Arity: &sema.Arity{Min: 2, Max: 3},
```

Reply:
It won't build unless we merge the latest master in `feature/range-type`. Can you do that please? I'll rebase on top of it and then update it here.

Reply:
Should be able to take a pull once https://github.com/onflow/cadence/pull/2747 has emerged.

Reply:
Pulled and updated now.